hal/src/nRF52840/interrupts_hal.cpp

```diff
@@ -141,25 +142,26 @@ void HAL_Interrupts_Attach(uint16_t pin, HAL_InterruptHandler handler, void* dat
         }
     } else if (err_code == NRFX_ERROR_NO_MEM) {
         // All channels have been used up
-        return;
+        return SYSTEM_ERROR_NO_MEMORY;
     }
 
     nrfx_gpiote_in_event_enable(nrf_pin, true);
+    return SYSTEM_ERROR_NONE;
 }
 
-void HAL_Interrupts_Detach(uint16_t pin) {
-    HAL_Interrupts_Detach_Ext(pin, 0, NULL);
+int HAL_Interrupts_Detach(uint16_t pin) {
+    return HAL_Interrupts_Detach_Ext(pin, 0, NULL);
 }
 
-void HAL_Interrupts_Detach_Ext(uint16_t pin, uint8_t keepHandler, void* reserved) {
+int HAL_Interrupts_Detach_Ext(uint16_t pin, uint8_t keepHandler, void* reserved) {
     if (keepHandler) {
         // This pin is used, don't detach it
-        return;
+        return SYSTEM_ERROR_INVALID_STATE;
     }
 
     Hal_Pin_Info* PIN_MAP = HAL_Pin_Map();
     if (PIN_MAP[pin].exti_channel == EXTI_CHANNEL_NONE) {
-        return;
+        return SYSTEM_ERROR_NOT_SUPPORTED;
```

Comment:
This should be `SYSTEM_ERROR_INVALID_STATE`.

Reply:
I fixed this one.

Reply:
Fixed